query_execution/QueryContext.cpp

```diff
@@ -58,7 +59,8 @@ QueryContext::QueryContext(const serialization::QueryContext &proto,
                            const CatalogDatabaseLite &database,
                            StorageManager *storage_manager,
                            const tmb::client_id scheduler_client_id,
-                           tmb::MessageBus *bus) {
+                           tmb::MessageBus *bus)
+    : database_(static_cast<const CatalogDatabase&>(database)) {
```

Comment:
Alternatively, `QueryManagerBase` class keeps track of the global memory usage, and `(Rebuild)WorkOrderCompleteMessge` contains the local memory usage for its resource, i.e., a block or a hash table. Thus, `Foreman` or `PolicyEnforcer` could sum up the memory usage in `QueryManageBase`.